File: performance/arithmetic_operations.py
import random

def main():
    random.seed(42)
    result = 0

    # Arithmetic operations benchmark
    for _ in range(100000):
        a = random.randint(1, 100)
        b = random.randint(1, 100)

        # Mix of operations to test different arithmetic paths
        sum = a + b
        diff = a - b
        prod = a * b
        quotient = a // (b % 10 + 1)

        result = result + sum
        result = result + diff
        result = result + prod
        result = result + quotient

    print(result)

main()

File: performance/arithmetic_operations.lua
math.randomseed(42)
local result = 0

for i = 0, 99999 do
  local a = math.random(1, 100)
  local b = math.random(1, 100)

  local sum = a + b
  local diff = a - b
  local prod = a * b
  local divisor = (b % 10) + 1
  local quotient = math.floor(a / divisor)

  result = result + sum
  result = result + diff
  result = result + prod
  result = result + quotient
end

print(result)
